### DISCUSSION TITLE
Can I custom a view url with other parameter different from the id?

### DISCUSSION THREAD

**Discussion by jals-es** [filamentphp/filament#3975] [Help] [answered]
I have posts. In DB they are identified by autoincrement id, and the default url to access to viewpost is using this id.
Example: `https://domain.com/posts/2`

I have a slug in posts and I want to change de view url to the slug.
Example: `https://domain.com/posts/slug-of-the-post`

Is it posible? How can I do that?

**Comment by danharrin** [comment, marked as answer]
On the resource set `protected static ?string $recordRouteKeyName = 'slug';`

**Reply by jals-es** [reply]
Thanks, this works well for the page, but the ViewAction in the table continue using the id. I can do something with ViewAction, or I have to make a custom action?

**Reply by danharrin** [reply]
What do you mean that the action uses it?

**Reply by jals-es** [reply]
When I set `protected static ?string $recordRouteKeyName = 'slug';` the view url changes to `https://domain.com/posts/slug-of-the-post` but the link in the table generated by the `Tables\Actions\ViewAction::make()` have the link in the id format `https://domain.com/posts/2`

**Reply by danharrin** [reply]
You probably need to change the route of that page (in `getPages()`) to use `{record:slug}` instead of `{record}`

**Reply by jals-es** [reply]
Thanks, I got it that way
